torch/distributed/_functional_collectives.py

```diff
@@ -517,3 +517,31 @@ def _register_ops():
     _register_ops()
 else:
     warnings.warn("PyTorch Distributed functional collectives do not work with torch::deploy.")
+
+# We allow torchdynamo to convert calls from legacy inplace APIs into traceable APIs
+# via a pseudo-inplace version (like a decomp) that uses the functional collective
+# and a copy.
+#
+# These schemas intentionally match torch.distributed.distributed_c10d.* ops that we are trying to remap from
+def all_gather_tensor_inplace(
+    output: torch.Tensor,
+    input: torch.Tensor,
+    group,  # TODO add a type,
+    async_op: bool = False,
+    tag: str = "",
+    gather_dim: int = 0
+):
+    assert not async_op, "Can't remap async version of inplace op to functional collective"
+    return output.copy_(all_gather_tensor(input, gather_dim, group, tag))
```

Comment:
I don't really understand how this fixes the need to define how to trace PG?

Reply:
PG comes into this function as a UserDefinedObjectVariable (iirc.  or whatever).

inside all_gather_tensor, it gets used in `expand_group` and is used as a key in a dict.  From there it turns into ints and a string, which are new constants we use in the actual traceable ops.

Reply:
Ok, we ended up adding ProcesGroupVariable to handle other cases (e.g. pg.rank() .size() calls), but PGVariable still isn't a critical part of how this rewrite part works.